### PR TITLE
Refactor clients and make facebook client testable

### PR DESCRIPTION
**Pull request by herman-rogers**
- Refactor clients to be more modular (for future use with steam implementation)

- Create a facebook and server client

- Wrap core facebook functionality in tests

- Make authentication configuration more modular (previously hardcoded in facebook_client)

- Create nested routing

### CODE REVIEW

File: api/client/facebook_client.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+		URL.RawQuery = parameters.Encode()
+
+		stringURL := URL.String()
+		http.Redirect(w, r, stringURL, http.StatusTemporaryRedirect)
+	})
+}
+
+// HandleFacebookCallback is the callback response after requesting authentication to facebook's api
+func (cli *FacebookClient) HandleFacebookCallback() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (herman-rogers) [inline api/client/facebook_client.go]: This is the main functionality that is 100% covered by tests.

Comment (SuperIzzo) [inline api/client/facebook_client.go]: oh nice, yeah it is quite big :D

File: api/client/server_client_test.go [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
 	w := httptest.NewRecorder()
-	cli.Handler.ServeHTTP(w, r)
+	router.ServeHTTP(w, r)
 
 	if w.Code != http.StatusOK {
 		t.Errorf("Expected %v, but got %v", http.StatusOK, w.Code)
 	}
 
-	resp := Response{}
+	resp := ServerResp{}
```
Comment (SuperIzzo) [inline api/client/server_client_test.go]: can this be ServerResponse?

File: api/client/server_client.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/WestCoastOpenSource/GameStore/pkg/logger"
+)
+
+// ServerClient is the client for interacting with the server
+type ServerClient struct {
```
Comment (SuperIzzo) [inline api/client/server_client.go]: NetworkClient maybe?

**Pull request merged**